app/register_validation.py

```diff
@@ -0,0 +1,140 @@
+    
+def validate_email(email):
+
+    email_pattern = r"((.*)@(.*))"
+           
+    # print(re.match(email_pattern, email, re.VERBOSE).groups())
+
+    match = re.match(email_pattern, email, re.VERBOSE)
+    
+    if not match:  
+        return False
+
+    email, local_part, domain = match.groups()
+
+    #CHECKING LOCAL PART OF AN EMAIL
+
+    #If unquoted, it may use any of these ASCII characters:
+
+    #uppercase and lowercase Latin letters A to Z and a to z
+    #digits 0 to 9
+    #printable characters !#$%&'*+-/=?^_`{|}~
+    #dot ., provided that it is not the first or last character and provided also that it does not appear consecutively (e.g., John..Doe@example.com is not allowed)
+    
+    #If quoted, it may contain Space, Horizontal Tab (HT), any ASCII graphic except Backslash and Quote and a quoted-pair consisting of a Backslash followed by HT, Space or any ASCII graphic; it may also be split between lines anywhere that HT or Space appears. In contrast to unquoted local-parts, the addresses ".John.Doe"@example.com, "John.Doe."@example.com and "John..Doe"@example.com are allowed.
+    
+    if not re.match(r"^\".+\"$", local_part):
+        
+        # re.search(r"(^\.|\.$)") ??
+        if re.search(r"^\.", local_part) or re.search(r"\.$", local_part):
+            return False
+    
+        if re.search(r"\.{2,}", local_part):
+            return False
+
+        if not re.match(r"^[A-Za-z0-9!#$%&'*+-/=?^_`{|}~\.]{1,256}$", local_part):
+            return False
+
+    else:
+        if not re.match(r"^[A-Za-z0-9!#$%&'*+-/=?^_`{|}~\.\s\"@]{1,256}$", local_part):
+            return False
+    
+
+    #CHECKING DOMAIN PART OF AN EMAIL
+
+    #uppercase and lowercase Latin letters A to Z and a to z;
+    #digits 0 to 9, provided that top-level domain names are not all-numeric;
+    #hyphen -, provided that it is not the first or last character.
+
+    not_allowed_domains = ["web"]
+
+
+    if not re.match(r"([A-Za-z0-9-\.]{1,63})", domain):
+        return False
+
+    if re.search(r"^\-", domain) or re.search(r"\-$", domain):
+        return False
+
+    if not re.search(r"\.", domain):
+        return False
+
+    if re.search(r"\.{2,}", domain):
+            return False
+
+    if re.search(r"\s", domain):
+        return False
+
+    if re.search(r"\.$", domain):
+        return False
+
+    for not_allowed_domain in not_allowed_domains:
+
+        if re.search(fr".{not_allowed_domain}$", domain):
+            return False
+
+    return True
+
+
+# TODO needs to be refactored
+def validate_date_format(date):
+    
+    pattern = r"^(\d\d\d\d)-(\d\d)-(\d\d)$"
+
+    groups = re.match(pattern, date)
+
+    if not groups:
+        return False
+    
+    year = int(groups.group(1))
+    month = int(groups.group(2))
+    day = int(groups.group(3)) 
+    
+    #TODO Should days in months to be checked? for example April 31 it's not correct.
+    #TODO to validate days in months
+    current_date = datetime.today()
```

Comment:
sorry, but realized just now it could be done easier. current_date is date object so to validate date would be to take the input date string and convert it to object, convertion will give you error if the date is incorrect, if convertion is sucessful just check if input_date < current_date :< and if year > 1900.

Reply:
I will do in the next PR

Reply:
solviing validation problem was big part of this PR solve it completely now imo